java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java

```diff
@@ -1850,7 +1850,23 @@ public boolean isCompileTimeConstantExpression(TreePath expression) {
         
         return ref.paramTypes;
     }
-    
+   
+    /**Check the var keyword in given statement line {@link Tree}.
+     * 
+     * @param leaf the leaf must be {@link Tree}
+     * @return the true if tree contains var keyword else return false.
+     */
+    public boolean isVarKeywordAvailable(@NonNull Tree leaf) {
```

Comment:
Please rename the method to say the intent what it is doing, not how - i.e. "isVarVariable" or something like that. The point of having this method is that we can change the implementation if a better way is found. Add @since and increment spec.version.base (currently is 2.30.0, should be 2.31.0) in java.source.base/nbproject/project.properties. The parameter should either be VariableTree or a TreePath.